Userland/Libraries/LibWeb/CSS/Length.cpp

```diff
@@ -39,6 +43,81 @@ float Length::relative_length_to_px(const Layout::Node& layout_node) const
     }
 }
 
+float Length::resolve_calculated_value(const Layout::Node& layout_node, float reference_for_percent) const
+{
+    // FIXME: Make this more error-friendly.
+    auto resolve_calc_value = [&layout_node, &reference_for_percent](CalculatedStyleValue::CalcValue const& calc_value) -> float {
+        if (calc_value.has<float>())
+            return calc_value.get<float>();
+
+        if (!calc_value.has<Length>())
+            VERIFY_NOT_REACHED();
+
+        auto length = calc_value.get<Length>();
+
+        return length.resolved_or_zero(layout_node, reference_for_percent).to_px(layout_node);
+    };
+
+    auto resolve_calc_number_value = [](CalculatedStyleValue::CalcNumberValue const& number_value) {
+        // FIXME: Allow for nested calc()
+        if (number_value.has<CalculatedStyleValue::CalcNumberSum>())
+            TODO();
+
+        if (!number_value.has<float>())
+            VERIFY_NOT_REACHED();
+
+        auto float_number = number_value.get<float>();
```

Comment:
Please consider `Variant::has<T>()`->`Variant::get<T>()` a bad thing to do, doing so will check the index twice.
Instead, use `Variant::visit()` if you're interested in handling many of the alternatives, or `Variant::get_pointer<T>()` if you're interested in only one of the alternatives.